independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/classloading/QuarkusClassLoader.java

```diff
@@ -241,13 +239,15 @@ public Enumeration<URL> getResources(String unsanitisedName, boolean parentAlrea
         }
         //TODO: in theory resources could have been added in dev mode
         //but I don't thing this really matters for this code path
+        Set<URL> resources = new LinkedHashSet<>();
         ClassPathElement[] providers = state.loadableResources.get(name);
         if (providers != null) {
             for (ClassPathElement element : providers) {
                 ClassPathResource res = element.getResource(name);
                 //if the requested name ends with a trailing / we make sure
                 //that the resource is a directory, and return a URL that ends with a /
                 //this matches the behaviour of URLClassLoader
+                boolean endsWithTrailingSlash = unsanitisedName.endsWith("/");
```

Comment:
Is it really an improvement to do it in the for loop?

Reply:
No, I'll move it out of the loop